File: crates/mct-lang-js-ts/tests/fixtures/webapp/invoice.ts
import { log } from "./logger";

const { add } = require("./mathUtils");

interface Priced {
    total: number;
}

const computeTotal = (price: number, taxRate: number): number => {
    return add(price, price * taxRate);
};

export class Invoice implements Priced {
    total = 0;

    addItem(price: number, taxRate: number): number {
        this.total = computeTotal(price, taxRate);
        log("item added");
        return this.total;
    }
}

File: crates/mct-lang-kotlin/tests/fixtures/billing-app/Invoice.kt
package billing

data class LineItem(val name: String, val price: Double)

class Invoice {
    var total: Double = 0.0

    fun addItem(item: LineItem): Double {
        total += item.price
        Logger.log("item added")
        return total
    }

    fun addItemWithTax(item: LineItem, taxRate: Double): Double {
        total += item.price * (1 + taxRate)
        Logger.log("item added with tax")
        return total
    }
}

fun Invoice.describe(): String = "total=$total"
